src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -104,26 +143,62 @@ class Poco
         public static void FirstGenericArgNotStringFail()
         {
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Parse<Dictionary<int, int>>(@"{""Key1"":1}"));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Parse<ImmutableDictionary<int, int>>(@"{""Key1"":1}"));
         }
 
         [Fact]
         public static void DictionaryOfList()
         {
             const string JsonString = @"{""Key1"":[1,2],""Key2"":[3,4]}";
 
-            IDictionary<string, List<int>> obj = JsonSerializer.Parse<IDictionary<string, List<int>>>(JsonString);
+            {
+                IDictionary<string, List<int>> obj = JsonSerializer.Parse<IDictionary<string, List<int>>>(JsonString);
 
-            Assert.Equal(2, obj.Count);
-            Assert.Equal(2, obj["Key1"].Count);
-            Assert.Equal(1, obj["Key1"][0]);
-            Assert.Equal(2, obj["Key1"][1]);
-            Assert.Equal(2, obj["Key2"].Count);
-            Assert.Equal(3, obj["Key2"][0]);
-            Assert.Equal(4, obj["Key2"][1]);
+                Assert.Equal(2, obj.Count);
```

Comment:
nit: seems like this could be put into a nested helper method here to avoid duplication.